tools/delete_mri_upload.pl

```diff
@@ -0,0 +1,578 @@
+}
+
+my $uploadID = $ARGV[0];
+
+#======================================#
+# Validate all command line arguments  #
+#======================================#
+if (!$ENV{LORIS_CONFIG}) {
+	print STDERR "\n\tERROR: Environment variable 'LORIS_CONFIG' not set\n\n";
+	exit $NeuroDB::ExitCodes::INVALID_ENVIRONMENT_VAR; 
+}
+
+if (!-e "$ENV{LORIS_CONFIG}/.loris_mri/$profile") {
+    print $Help; 
+    print STDERR "Cannot read profile file '$ENV{LORIS_CONFIG}/.loris_mri/$profile'\n";  
+    exit $NeuroDB::ExitCodes::PROFILE_FAILURE;
+}
+
+# Incorporate contents of profile file
+{ package Settings; do "$ENV{LORIS_CONFIG}/.loris_mri/$profile" }
+
+if ( !@Settings::db ) {
+    print STDERR "ERROR: You don't have a \@db setting in file "
+                 . "$ENV{LORIS_CONFIG}/.loris_mri/$profile";
+    exit $NeuroDB::ExitCodes::DB_SETTINGS_FAILURE;
+}
+
+#==================================#
+# Establish database connection    #
+#==================================#
+my $dbh = &NeuroDB::DBI::connect_to_db(@Settings::db);
+
+my $dataDirBasepath    = NeuroDB::DBI::getConfigSetting(\$dbh,'dataDirBasepath');
+$dataDirBasepath =~ s!/$!!;
+my $tarchiveLibraryDir = NeuroDB::DBI::getConfigSetting(\$dbh,'tarchiveLibraryDir');
+$tarchiveLibraryDir =~ s!/$!!;
+
+my $query = "SELECT m.TarchiveID, t.ArchiveLocation, m.SessionID "
+    .       "FROM mri_upload m "
+    .       "JOIN tarchive t USING (TarchiveID) "
+    .       "WHERE m.UploadID = ?";
+my $sth = $dbh->prepare($query);
+$sth->execute($uploadID);
+my($tarchiveID, $archiveLocation, $sessionID) = $sth->fetchrow_array();
+
+if($sth->rows == 0) {
+	print STDERR "No upload found in table mri_upload with upload ID '$uploadID'\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(!$tarchiveID) {
+	print STDERR "No tarchive ID found in table mri_upload for upload with ID $uploadID\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+if(!defined $tarchiveLibraryDir && $archiveLocation !~ /^\//) {
+    print STDERR "Cannot determine absolute path for archive '$archiveLocation' "
+        . "since config setting 'tarchiveLibraryDir' is not set. Aborting.\n";
+    exit $NeuroDB::ExitCodes::MISSING_CONFIG_SETTING;
+}
+
+$archiveLocation = "$tarchiveLibraryDir/$archiveLocation" unless $archiveLocation =~ /^\//;
+
+#===================================================#
+# Make sure there are no entries in files_qc_status #
+# and feedback_mri_comments for the files in the    #
+# archive                                           #
+#===================================================#
+
+if(&hasQcOrComment($dbh, $uploadID, $tarchiveID)) {
+	print STDERR "Cannot delete upload '$uploadID': there is QC information"
+        . " associated to that archive\n";
+	exit $NeuroDB::ExitCodes::INVALID_ARG;
+}
+
+#===========================================#
+# Backup all files found in                 #
+# tables files and files_intermediary for   #
+# the specified upload_id                   #
```

Comment:
should we also backup the tarchive? (and maybe the files in trashbin?)

Reply:
The tarchive is backed up. I''l update the script so it backs up the files in `trashbin`.